meetings/2022-11/dec-01.md

```diff
@@ -0,0 +1,1104 @@
+
+JWK: That's a good suggestion. Thank you. I will try to bring this topic up in the research call.
+
+BT: Sorry newt. there's one more item on the Queue, RPY . do you want to go ahead?
+
+RPY: Kind of, a continuation of what SFC was saying, and so my own feeling. So if the goal here is to hide the complexity of promises then I think this is to surface level because it's really just renaming utilities that are there. I think maybe explore where users are finding that complexity and what more we could do to make it easier for people to not have to be exposed to promises. Because right now, like as soon as you debug these objects or anything you're going to be exposed to promises.
+
+JWK: I think there isn't much space we can further hide the promises, because for things already async you don't need a promise but if things are not, for example from event emitter it has to call the promise constructor.
+
+RPY: Okay, in that case, you may as well deliberately force users to learn promises and understand how to use them,
+
+JWK: Hmm. Okay. Thanks.
+
+JHD: I have a comments about that. So for me, this proposal is not about hiding promises in any way.  I'm technically a co-champion this proposal and I think that that motivation is actually a reason not to have it. For me it's about the ergonomics of using promises, and reaching await and being able to use await.all For example, instead of having to kind of break the chain, when you do promise.all you have to do the thing with the parentheses that admittedly with something like pipeline, you would it would be a more ergonomic. My pipeline doesn't exist, yet to me this is like the same. A lot of the foot guns that I see in a way async await usage are caused by the attractiveness of using await. Unnecessarily. And when these options syntactic options are available than they become equally attractive and often they are the more correct Choice than await.
+
+JWK: await an array?
+
+JHD Yeah, I mean if you awake.all in a ray that is much more ergonomic and attractive than like a. So in other words what the people will do like `await a; await b; await c`, and store those in variables, what they should be doing if a, b and c depend on each other is await.all like like an array you know using a array destructuring equals await.all of an array of a promise to get out the results. Because that's you know, concurrent as and there's no there's no reason otherwise for them to be cereal. and that using awake got all is far more ergonomic and appealing like to the point where I think more users will do it rather than unnecessarily serializing their asynchronous functions. you know, like there. very sync sync functions. so, to me, that is the primary and most important aspect of this proposal. it like hiding promises. I think is very much an anti goal. Nobody should ever forget how common that promises is exist, but it should be like this to me this this avoids very, very frequent footguns.
```

Comment:
```suggestion
JHD: Yeah, I mean if you `await.all` an array, that is much more ergonomic and attractive than … in other words, when people do `await a; await b; await c`, and store those in variables, what they should be doing if `a`, `b`, and `c` depend on each other is using array destructuring equals `await.all` of an array of a promise to get out the results. Because that's concurrent,and there's no reason otherwise for them to be serial. and that using `await.all` is far more ergonomic and appealing to the point where I think more users will do it rather than unnecessarily serializing their asynchronous functions. So, to me, that is the primary and most important aspect of this proposal. Hiding promises I think is very much an anti goal. Nobody should ever forget that Promises exist, but to me this this avoids very, very frequent footguns.
```